compute_sdk/globus_compute_sdk/sdk/login_manager/tokenstore.py

```diff
@@ -36,22 +36,31 @@ def invalidate_old_config() -> None:
 
 
 def ensure_compute_dir() -> pathlib.Path:
+    user_dirname_str = os.getenv("GLOBUS_COMPUTE_USER_DIR")
+    if user_dirname_str:
+        user_dirname = pathlib.Path(user_dirname_str)
+        if user_dirname.is_dir():
+            return user_dirname
+        elif user_dirname.is_file():
+            raise FileExistsError(f"{user_dirname} must be a directory, not a file")
+        else:
+            raise FileNotFoundError(
+                f"{user_dirname} could not be found on the file system"
```

Comment:
I feel like if we only intend for the user to be able to use an existing dir when specifying GLOBUS_COMPUTE_USER_DIR, then we should be more clear in the error message what we expect them to do.  ie. Make sure it's an existing directory, that we can not create a directory from GLOBUS_COMPUTE_USER_DIR if it didn't exist, etc.  Namely, if the person running it isn't the person who set the ENV var, (or forgot that he did), the message reminds them what the root cause is (a custom thing is set but not ready to use and we won't auto-create as we would the default .globus_compute).

ie. This var is a new feature that we can document using the changelog, but when people try to use it in the future further removed from reading release notes, give more context in case of failure.

Reply:
Good thought, but I made some [adjustments](https://github.com/funcx-faas/funcX/pull/1125#discussion_r1181978213) that make it unnecessary to use different error messages.